hazelcast/src/main/java/com/hazelcast/map/impl/event/MapEventPublisherImpl.java

```diff
@@ -265,6 +265,11 @@ public void hintMapEvent(Address caller, String mapName, EntryEventType eventTyp
         // NOP
     }
 
+    @Override
+    public boolean hasEventListener(String mapName) {
+        return eventService.hasEventRegistration(SERVICE_NAME, mapName);
```

Comment:
Good idea but if we go that way, we need to iterate over all registered map-listeners to find if there is any `includeValue=true` registration exists or not.